### ISSUE TITLE
port forwarding not working with postman

### ISSUE THREAD

**Issue by u000fsr** [MicrosoftDocs/vsonline#418]
the port forwarding works ok in the browser for viewing pages, but when using something like postman for testing an api it's not working out.

**Comment by jkeech** [comment]
@u000fsr, this is expected when connecting from the browser currently because the remote side is not exposed on any publicly accessible endpoint. The url that shows up in the browser is actually intercepted by a local service worker and proxied to the remote side over the secure connection to the environment. So hitting that url from outside of the browser won't work.

We are building a different port-forwarding service that is not based on service workers, and in that case it will run on a real publicly accessible endpoint, although it will still be locked down with auth by default. We also plan to add some gestures to enable opening it up to not require a special auth header, but the general principle is to keep things secure by default so only you can access your environment and the code running inside of it unless you specifically choose to share it with others.

If you connect from VS Code on your desktop, our extension has the ability to spin up a local webserver on whatever port you have forwarded, which then proxies the request to the environment in the same way that the service worker does in the browser. The main difference is that is it actually listening on port 3000 locally (or whatever port you forwarded). This means you can use any other tool on your desktop (postman, curl, etc) and hit localhost:3000, which will be proxied to the remote side.

In short, the port forwarding experience is more complete currently when accessing environments from VS Code on the desktop instead of in the browser, but we are working to fill that gap and make it just as good in the browser as it is on the desktop.

**Comment by u000fsr** [comment]
Great, thanks for the info!
